src/manifolds/FixedRankMatrices.jl

```diff
@@ -408,3 +421,34 @@ function zero_tangent_vector!(
     X.Vt .= zeros(eltype(X.Vt), k, n)
     return X
 end
+
+@doc raw"""
+    embed(p::SVDMPoint)
+
+Embed the point `p` from its `SVDMPoint` representation.
+"""
+
+function embed!(::FixedRankMatrices, q, p)
+    return q .= p.U * Diagonal(p.S) * p.Vt
+end
+function embed(M::FixedRankMatrices{m,n,k}, p) where {m,n,k}
+    q = zeros(m, n)
+    return embed!(M, q, p)
+end
+
+@doc raw"""
+    embed(::FixedRankMatrices, Y, p, X)
+
+Embed the tangent vector `X` at point `p` in `M` from
+its `UMVTVector` representation.
+"""
+function embed!(::FixedRankMatrices, Y, p, X)
+    Y = p.U * X.M * p.Vt
+    mul!(Y, X.U, p.Vt, true, true)
+    return mul!(Y, p.U, X.Vt, true, true)
```

Comment:
I see now we can eliminate one matrix product. Also, we need to mutate the passed `Y`
```suggestion
    tmp = p.U * X.M
    tmp .+= X.U
    mul!(Y, tmp, p.Vt)
    return mul!(Y, p.U, X.Vt, true, true)
```